From 6e5b1852be0247803ea070a6708010a0d3c2bc3e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Thu, 19 Dec 2019 22:39:11 +0100
Subject: [PATCH] init: quote $rootfstype during comparison

Without this change this may result in a warning message to be printed
during boot if the rootfstype kernel parameter is not set.
---
 initramfs-init.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/initramfs-init.in b/initramfs-init.in
index 76f284c..eaacb13 100755
--- a/initramfs-init.in
+++ b/initramfs-init.in
@@ -503,7 +503,7 @@ if [ -n "$KOPT_root" ]; then
 		mkdir -p /media/root-rw/work /media/root-rw/root
 		mount -t overlay -o lowerdir=/media/root-ro,upperdir=/media/root-rw/root,workdir=/media/root-rw/work overlayfs $sysroot
 	else
-		if [ $rootfstype = "zfs" ]; then
+		if [ "$rootfstype" = "zfs" ]; then
 			prepare_zfs_root
 		fi
 		mount ${rootfstype:+-t} ${rootfstype} \
